fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewPartitionFunctionChecker.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.analyzer;
+
+import com.google.common.collect.Maps;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.catalog.PrimitiveType;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.starrocks.sql.common.TimeUnitUtils.TIME_MAP;
+
+public class MaterializedViewPartitionFunctionChecker {
+    @FunctionalInterface
+    public interface CheckPartitionFunction {
+        boolean check(Expr expr);
+    }
+
+    public static final Map<String, CheckPartitionFunction> FN_NAME_TO_PATTERN;
+
+    static {
+        FN_NAME_TO_PATTERN = Maps.newHashMap();
+        // can add some other functions
+        FN_NAME_TO_PATTERN.put("date_trunc", MaterializedViewPartitionFunctionChecker::checkDateTrunc);
+        FN_NAME_TO_PATTERN.put("time_slice", MaterializedViewPartitionFunctionChecker::checkTimeSlice);
```

Comment:
In the future, we will support many time functions, for each function, do we must  write so complex check code respectively?

Reply:
I found `time_slice ` is not supported yet before MV's create partition uses the `automatic partitioning`. so I remove `time_slice` .